gobblin-data-management/src/main/java/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -147,11 +156,18 @@ private void publishFileSet(CopyEntity.DatasetAndPartition datasetAndPartition,
         datasetWorkUnitStates.iterator().next().getProp(CopySource.SERIALIZED_COPYABLE_DATASET));
     Path datasetWriterOutputPath = new Path(this.writerOutputDir, datasetAndPartition.identifier());
 
-    log.info(String.format("Publishing fileSet from %s for dataset %s", datasetWriterOutputPath,
-        metadata.getDatasetURN()));
+    log.info(String.format("[%s] Publishing fileSet from %s for dataset %s", datasetAndPartition.identifier(),
+        datasetWriterOutputPath, metadata.getDatasetURN()));
 
+    List<CommitStep> prePublish = getCommitSequence(datasetWorkUnitStates, PrePublishStep.class);
+    List<CommitStep> postPublish = getCommitSequence(datasetWorkUnitStates, PostPublishStep.class);
+    log.info(String.format("[%s] Found %d prePublish steps and %d postPublish steps.", datasetAndPartition.identifier(),
+        prePublish.size(), postPublish.size()));
+
+    executeCommitSequence(prePublish);
```

Comment:
It can cause any of that, same as any failed publish step.
I am using `CommitStep` here with the objective of eventually using exactly-once semantics (very easy to assemble a `CommitSequence` if we already have the steps). However, I understand exactly-once is not ready yet.